pkg/rlptools/wasm_utils.go

```diff
@@ -20,7 +21,7 @@ var (
 	WASMFilterImageURL = common.FetchEnv("RELATED_IMAGE_WASMSHIM", "oci://quay.io/kuadrant/wasm-shim:latest")
 )
 
-// WasmRules computes WASM rules from the policy and the targeted Route (which can be nil when a gateway is targeted)
+// WasmRules computes WASM rules from the policy and the targeted route.
 func WasmRules(rlp *kuadrantv1beta2.RateLimitPolicy, route *gatewayapiv1beta1.HTTPRoute) []wasm.Rule {
```

Comment:
This method should return `error` as well. One of the use cases I found is when the route selector does not match any HTTPRoute rule (route)

Reply:
As an alternative, the route selectors validity could be evaluated in a early stage of the reconcilliation loop to report in the status when the selector does not select any route. With this approach, the `WasmRules` only returns rules

Reply:
Right now, it's simply skipping a route selector that does not select anything.

Instead of raising a error here, I'd advocate for a combination of the following:
1. the alternative you suggest – i.e. checking for route selector mismatches early on; I'd say in a validating webhook;
2. reporting in the status of the RLP about all limits that will not be applied due to route selector mismatch. This one because this reconciliation can be triggered by changes in the HTTPRoutes as well.